libs/api/domains/license-service/src/lib/licenseService.service.ts

```diff
@@ -624,21 +634,51 @@ export class LicenseServiceService {
     data: string,
   ): Promise<VerifyLicenseBarcodeResult> {
     if (isJWT(data)) {
-      return this.getDataFromToken(data)
+      // Verify the barcode data as a token, e.g. new barcode format
+      const tokenData = await this.getDataFromToken(data)
+
+      return {
+        barcodeType: VerifyLicenseBarcodeType.V2,
+        ...tokenData,
+      }
     }
 
+    // else fallback to the old barcode format
+
     if (!isJSON(data)) {
       this.logError('Invalid JSON data')
 
-      return COMMON_VERIFY_ERROR
+      return {
+        barcodeType: VerifyLicenseBarcodeType.UNKNOWN,
+        ...COMMON_VERIFY_ERROR,
+      }
     }
 
-    const { passTemplateId }: { passTemplateId?: string } = JSON.parse(data)
+    const {
+      passTemplateId,
+      expires,
+      date,
+    }: { passTemplateId?: string; expires?: string; date?: string } =
+      JSON.parse(data)
 
     if (!passTemplateId) {
       this.logError('No passTemplateId found in data')
 
-      return COMMON_VERIFY_ERROR
+      return {
+        barcodeType: VerifyLicenseBarcodeType.UNKNOWN,
+        ...COMMON_VERIFY_ERROR,
+      }
+    }
+
+    const parsedExpireTime = expires || date
+
+    // If the expiration date is in the past, the barcode is expired
+    if (parsedExpireTime && !isAfter(new Date(parsedExpireTime), new Date())) {
+      return {
+        barcodeType: VerifyLicenseBarcodeType.PK_PASS,
+        valid: false,
+        error: VerifyLicenseBarcodeError.EXPIRED,
+      }
```

Comment:
Isn't the expiration already being checked elsewhere on the server? Weird to check twice if the pkpass data is expired.

If not, we need to add `licenseType` also to this response to support current app design.

Reply:
No I checked the smartApi just throws the error if date is expired. 
<img width="1010" alt="image" src="https://github.com/island-is/island.is/assets/112904566/0543cd86-cc03-4ef1-b740-4d261002cead">

Reply:
And here with the error check
<img width="289" alt="image" src="https://github.com/island-is/island.is/assets/112904566/bb636425-3ded-40c4-92fa-c3cff2a9dad3">

Reply:
And now with the licenseType :)
<img width="334" alt="image" src="https://github.com/island-is/island.is/assets/112904566/18161f73-2ad0-4475-ac97-dc7f3439d558">